pkg/ingester/ingester.go

```diff
@@ -1201,6 +1229,8 @@ func (i *Ingester) Push(ctx context.Context, req *cortexpb.WriteRequest) (*corte
 
 			case errors.As(cause, &errMaxSeriesPerLabelSetLimitExceeded{}):
 				perLabelSetSeriesLimitCount++
+				// We only track per labelset discarded samples for throttling by labelset limit.
+				reasonCounter.increment(matchedLabelSetLimits, perLabelsetSeriesLimit)
```

Comment:
Even though we are only tracking `perLabelsetSeriesLimit`, we still wanna have this reasonCounter struct so we can easily add other reasons in the future, right?